api/_utils/open-weather-api.ts

```diff
@@ -90,6 +136,36 @@ export class OpenWeather {
 
       if (json.cod !== statusCodeOk) throw new Error('API response not OK');
 
+      return json;
+    };
+
+  fetchForecast =
+    async (params: OWMParams = {}): Promise<OWMForecastWeatherData> => {
+      const url = new URL('https://api.openweathermap.org/data/2.5/forecast');
+      const searchParams = new URLSearchParams();
+      /* eslint-disable-next-line no-invalid-this */
+      searchParams.set('appid', this.apiKey);
+
+      for (const [param, value] of Object.entries(params)) {
+        if (value !== '') searchParams.set(param, value);
+      }
+
+      url.search = searchParams.toString();
+      const response = await fetch(url.toString());
+
+      if (!response.ok) {
+        throw Object.assign(new Error(), {
+          message: 'Fetch response not OK',
+          name: 'FetchError',
+          response,
+        });
+      }
+
+      const json: OWMForecastWeatherData = await response.json();
+      const statusCodeOk = 200;
+
+      if (json.cod !== statusCodeOk) throw new Error('API response not OK');
```

Comment:
Guess what? The API returns the `cod` value in _this_ response as a `string`. 😠

This entire API is very poorly designed.